diff --git a/Makefile b/Makefile
index 77886dd..38b802a 100644
--- a/Makefile
+++ b/Makefile
@@ -40,7 +40,7 @@ endif
 #CONFIG_COSMO=y
 
 # installation directory
-PREFIX?=/usr/local
+PREFIX?=../
 
 # use the gprof profiler
 #CONFIG_PROFILE=y
@@ -232,7 +232,7 @@ QJS_LIB_OBJS=$(OBJDIR)/quickjs.o $(OBJDIR)/dtoa.o $(OBJDIR)/libregexp.o $(OBJDIR
 QJS_OBJS=$(OBJDIR)/qjs.o $(OBJDIR)/repl.o $(QJS_LIB_OBJS)
 
 HOST_LIBS=-lm -ldl -lpthread
-LIBS=-lm
+LIBS=-lm -latomic
 ifndef CONFIG_WIN32
 LIBS+=-ldl -lpthread
 endif
@@ -242,22 +242,22 @@ $(OBJDIR):
 	mkdir -p $(OBJDIR) $(OBJDIR)/examples $(OBJDIR)/tests
 
 qjs$(EXE): $(QJS_OBJS)
-	$(CC) $(LDFLAGS) $(LDEXPORT) -o $@ $^ $(LIBS)
+	$(CC) $(LDFLAGS) $(LDEXPORT) -o $@ $^ $(LIBS) $(CFLAGS)
 
 qjs-debug$(EXE): $(patsubst %.o, %.debug.o, $(QJS_OBJS))
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS) $(CFLAGS)
 
 qjsc$(EXE): $(OBJDIR)/qjsc.o $(QJS_LIB_OBJS)
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS) $(CFLAGS)
 
 fuzz_eval: $(OBJDIR)/fuzz_eval.o $(OBJDIR)/fuzz_common.o libquickjs.fuzz.a
-	$(CC) $(CFLAGS_OPT) $^ -o fuzz_eval $(LIB_FUZZING_ENGINE)
+	$(CC) $(CFLAGS_OPT) $^ -o fuzz_eval $(LIB_FUZZING_ENGINE) $(CFLAGS)
 
 fuzz_compile: $(OBJDIR)/fuzz_compile.o $(OBJDIR)/fuzz_common.o libquickjs.fuzz.a
-	$(CC) $(CFLAGS_OPT) $^ -o fuzz_compile $(LIB_FUZZING_ENGINE)
+	$(CC) $(CFLAGS_OPT) $^ -o fuzz_compile $(LIB_FUZZING_ENGINE) $(CFLAGS)
 
 fuzz_regexp: $(OBJDIR)/fuzz_regexp.o $(OBJDIR)/libregexp.fuzz.o $(OBJDIR)/cutils.fuzz.o $(OBJDIR)/libunicode.fuzz.o
-	$(CC) $(CFLAGS_OPT) $^ -o fuzz_regexp $(LIB_FUZZING_ENGINE)
+	$(CC) $(CFLAGS_OPT) $^ -o fuzz_regexp $(LIB_FUZZING_ENGINE) $(CFLAGS)
 
 libfuzzer: fuzz_eval fuzz_compile fuzz_regexp
 
@@ -320,7 +320,7 @@ $(OBJDIR)/fuzz_%.o: fuzz/fuzz_%.c | $(OBJDIR)
 	$(CC) $(CFLAGS_OPT) -c -I. -o $@ $<
 
 $(OBJDIR)/%.host.o: %.c | $(OBJDIR)
-	$(HOST_CC) $(CFLAGS_OPT) -c -o $@ $<
+	$(HOST_CC) -O1 -c -o $@ $<
 
 $(OBJDIR)/%.pic.o: %.c | $(OBJDIR)
 	$(CC) $(CFLAGS_OPT) -fPIC -DJS_SHARED_LIBRARY -c -o $@ $<
diff --git a/qjs.c b/qjs.c
index 2eaa9ee..2a0f439 100644
--- a/qjs.c
+++ b/qjs.c
@@ -287,7 +287,7 @@ static size_t get_suffixed_size(const char *str)
 
 void help(void)
 {
-    printf("QuickJS version " CONFIG_VERSION "\n"
+    printf("QuickJS version 2024-02-14\n"
            "usage: " PROG_NAME " [options] [file [args]]\n"
            "-h  --help         list options\n"
            "-e  --eval EXPR    evaluate EXPR\n"
diff --git a/qjsc.c b/qjsc.c
index f9e1928..47a659f 100644
--- a/qjsc.c
+++ b/qjsc.c
@@ -340,7 +340,7 @@ static const char main_c_template2[] =
 
 void help(void)
 {
-    printf("QuickJS Compiler version " CONFIG_VERSION "\n"
+    printf("QuickJS Compiler version 2024-02-14\n"
            "usage: " PROG_NAME " [options] [files]\n"
            "\n"
            "options are:\n"
diff --git a/quickjs-libc.c b/quickjs-libc.c
index 0788d8c..d161a16 100644
--- a/quickjs-libc.c
+++ b/quickjs-libc.c
@@ -47,13 +47,13 @@
 #include <sys/ioctl.h>
 #include <sys/wait.h>
 
-#if defined(__FreeBSD__)
+// #if defined(__FreeBSD__)
 extern char **environ;
-#endif
+// #endif
 
-#if defined(__APPLE__) || defined(__FreeBSD__)
+// #if defined(__APPLE__) || defined(__FreeBSD__)
 typedef sig_t sighandler_t;
-#endif
+// #endif
 
 #if defined(__APPLE__)
 #if !defined(environ)
diff --git a/quickjs.c b/quickjs.c
index 29c53ba..bec2d3f 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -6594,7 +6594,7 @@ void JS_ComputeMemoryUsage(JSRuntime *rt, JSMemoryUsage *s)
 
 void JS_DumpMemoryUsage(FILE *fp, const JSMemoryUsage *s, JSRuntime *rt)
 {
-    fprintf(fp, "QuickJS memory usage -- " CONFIG_VERSION " version, %d-bit, malloc limit: %"PRId64"\n\n",
+    fprintf(fp, "QuickJS memory usage -- 2024-02-14 version, %d-bit, malloc limit: %"PRId64"\n\n",
             (int)sizeof(void *) * 8, s->malloc_limit);
 #if 1
     if (rt) {
diff --git a/run-test262.c b/run-test262.c
index a42b9b5..31ac267 100644
--- a/run-test262.c
+++ b/run-test262.c
@@ -1987,7 +1987,7 @@ void run_test_dir_list(namelist_t *lp, int start_index, int stop_index)
 
 void help(void)
 {
-    printf("run-test262 version " CONFIG_VERSION "\n"
+    printf("run-test262 version 2024-02-14\n"
            "usage: run-test262 [options] {-f file ... | [dir_list] [index range]}\n"
            "-h             help\n"
            "-a             run tests in strict and nostrict modes\n"
